File: 프로그래머스/lv2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
    let queue = new Array(bridge_length).fill(0);
    let truck_weigth = 0, cnt = bridge_length;
    while(truck_weights.length){
        queue.shift();
        truck_weigth = 0;
        if(weight >= queue.reduce((sum, v) => sum + v, truck_weights[0]))
            truck_weigth = truck_weights.shift();   
        queue.push(truck_weigth);
        cnt++;
    }
    return cnt;
}

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.java
import java.util.*;

class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int answer = 0;
        int currentWeight = 0;
        Queue<Integer> queue = new LinkedList<>();
        
        for(int i=0; i<bridge_length; i++) {
            queue.offer(0);
        }

        int index = 0;
        while(index < truck_weights.length) {
            currentWeight -= queue.poll();
            
            if(currentWeight + truck_weights[index] > weight) {
                queue.offer(0);    
            } else {
                queue.offer(truck_weights[index]);
                currentWeight+=truck_weights[index];
                index++;
            }
            answer++;
        }
        
        
        return answer + bridge_length;
    }
}